docs/content/reference/03-middleware.md

```diff
@@ -154,219 +261,141 @@ module.exports = {
-   `my-domain.auth0.com`.
-- `secret`: **Required**. either in the config object or in the env var `OAUTH_CLIENT_SECRET`.
-   Provided by your oauth service when you registered your application.
-- `clientId`: **Required** either in the config object or in the env var `OAUTH_CLIENT_ID`. Provided
-   by your oauth service when you registered your application.
-- `userKey`: The key to delete from session storage on logout. A session key is *not set* by
-  middleware; you responsible for setting any session storage yourself. Defaults to `user`.
-- `callbackUrl`: A full URI, with protocol and domain. Read from the env var `OAUTH_CALLBACK_URL`;
-  defaults to the uri `/callback` on your app.
-- `tokenUrl`: A full URI, with protocol and domain. Read from the env var `OAUTH_TOKEN_URL`;
-  defaults to `https://${OAUTH_DOMAIN}/oauth/token`
-- `userinfoUrl`: A full URI, with protocol and domain. Read from the env var `OAUTH_USERINFO_URL`.
-  If no value is provided, derived from the `domain` parameter as `https://${domain}/userinfo`
-- `authorizationUrl`: A full URI, with protocol and domain. Read from the env var
-  `OAUTH_AUTHORIZATION_URL`; defaults to `https://${OAUTH_DOMAIN}/authorize`
-- `expiryLeewaySeconds`: Read from the env var `OAUTH_EXPIRY_LEEWAY`. Defaults to 60 seconds.
-- `defaultNextPath`: defaults to `/`
-- `logoutRoute`: defaults to `/logout`,
-- `returnTo`: A full URI, with protocol and domain. Read from the env var `OAUTH_LOGOUT_CALLBACK`;
-  defaults to the uri `/` on your app.
-- `logoutUrl`: Read from the env var `AUTH_LOGOUT_URL`. Defaults to
-  `https://${OAUTH_DOMAIN}/v2/logout`
+* * *
 
-The OAuth middleware has many configuration knobs and dials to turn, but the middleware is usable in
-development if you set three environment variables: `OAUTH_DOMAIN`, `OAUTH_CLIENT_SECRET`, and
-`OAUTH_CLIENT_ID`. If you set those variables, the code required to attach oauth middleware
-looks like this:
+### `template`
 
-```javascript
-const { middleware } = require('./boltzmann')
+The `template` middleware is available if you have enabled the templating feature with
+`--templates=on`.  It enables returning rendered [nunjucks](https://mozilla.github.io/nunjucks/)
+templates from handlers. See the [website features overview](@/concepts/03-websites.md) for a
+description of how to use templates to build websites and the development conveniences provided.
 
-// with process.env.{OAUTH_DOMAIN, OAUTH_CLIENT_SECRET, OAUTH_CLIENT_ID} all set
-module.exports = {
-  APP_MIDDLEWARE: [
-    middleware.oauth
-  ]
-};
-```
+#### Arguments
 
-**Advanced configuration**:
+-   `paths`: an array of string paths where template files are looked up; defaults to `./templates`, a
+    single directory relative to the application root.
+-   `filters`: an object specifying [custom
+    filters](https://mozilla.github.io/nunjucks/api#custom-filters) to add to the Nunjucks renderer.
+    Object keys are filter names, and the values must be filter functions.
```

Comment:
I believe we enhance the base nunjucks `filters` behavior by allowing `async` functions to be registered.

(We'll probably want to add examples of registering and using template filters and tags to the websites doc.)